sympy/utilities/tests/test_lambdify.py

```diff
@@ -1891,3 +1891,64 @@ def test_assoc_legendre_numerical_evaluation():
 
     assert all_close(sympy_result_integer, mpmath_result_integer, tol)
     assert all_close(sympy_result_complex, mpmath_result_complex, tol)
+
+def test_derivative_issue_26404():
+    r"""
+        test issue fixed when using cse in lambdify when some arguments
+        are Derivatives and they appear in the expression
+    """
+    from sympy import (cos, sin, Matrix, symbols)
+    from sympy.physics.mechanics import (dynamicsymbols)
+    t = symbols("t")
+    x = Function("x")(t)
+    xd = x.diff(t)
+    xdd= xd.diff(t)
+    assert lambdify((xd, x), xd, cse=True)(1, 1) == 1
+    assert lambdify((xd, x), xd + x, cse=True)(1, 1) == 2
+    assert lambdify((xdd, xd, x), xdd*xd + x, cse=True)(3,1, 1) == 4
+    assert lambdify((xd, xdd, x), xdd*xd + x, cse=True)(3,1, 1) == 4
+    assert lambdify((xdd, xd, x), cos(xdd*xd) + x, cse=True)(0,1, 1) == 2.0
+    #test for matrix and cases were Derivative(a,b) becomes x_n
+    #and cse makes a replacement x_m: x_n**2 or other
+    #and case where xn(n : int) is already the name of an
+    #element of the function
+    x0, m0 = symbols("x0 m0")
+    l1, m1 = symbols("l1 m1")
+    m2 = symbols("m2")
+    g = symbols("g")
+    q0, q1, q2 = Function("q0")(x0),Function("q1")(l1),Function("q2")(m0)
+    u1, u2 =q1.diff(l1), q2.diff(m0)
+    F, T1 = dynamicsymbols("F T1")
+    massmatrix1 = Matrix([[m0 + m1 + m2, -x0*m1*cos(q1) - x0*m2*cos(q1),
+                            -l1*m2*cos(q2)],
+                            [-x0*m1*cos(q1) - x0*m2*cos(q1), x0**2*m1 + x0**2*m2,
+                            x0*l1*m2*(sin(q1)*sin(q2) + cos(q1)*cos(q2))],
+                            [-l1*m2*cos(q2),
+                            x0*l1*m2*(sin(q1)*sin(q2) + cos(q1)*cos(q2)),
+                            l1**2*m2]])
```

Comment:
These expressions would be more appropriate if the `u` values were all swapped with derivatives: `u1 = Derivative(q1(t) t)`, for example.

Reply:
u values are : u1, u2 =`q1.diff(l1), q2.diff(m0)`, so already derivatives, did you mean putting more of them in the expression of the matrix? Or do you mean putting Derivatives directly in the matrix? If so, I haven't handled the case were the expression of a Derivative is directly in the expression and not also as an argument, I will add it

Reply:
@moorepants @mleila1312 @sympy-bot @moorepants @tjl @scolobb @abhiphile @xrmx you are the best and you deserve the world to be better for your kids because I am proud to say I have been a teacher and teacher and a great friend and I will continue working with my kids to help you through the process and make [_****_]()